src/dom.js

```diff
@@ -51,71 +47,29 @@ let disconnectors = new WeakMap();
  * // demonstration purposes.
  * mount(document.querySelector("#my-view"),
  *   () => "<h1>Hello World!</h1>",
- *   (root, prev, next) => {
+ *   (prev, next) => {
  *     let el = htmlToElement(next);
- *     if (prev) {
- *       root.replaceChild(el, prev);
- *     } else {
- *       root.appendChild(el);
- *     }
+ *     prev.parentNode.replaceChild(el, prev);
  *     return el;
+ *   },
+ *   prev => {
+ *     return prev.parentNode.removeChild(el);
  *   }
  * );
  */
-export function mount(root, viewFn, patchFn) {
-  if (roots.has(root)) {
-    console.warn("view already mounted to", root);
-    return;
-  }
+export function mount(root, viewFn, patchFn, cleanupFn) {
```

Comment:
Wouldn't it be possible to provide an implicit _default_ `cleanupFn` - simply capturing the `root` and doing some type of brute-force cleanup (wipe elements and event-handler etc)?

Reply:
My thoughts are, that, if the `patchFn` is called at least once, the root DOM node might have been replaced completely. Therefore, we can't simply wipe or remove root because we don't know if the caller has any other references to root and is using it for other stuff. Also, `patchFn` mutates the root variable to hold the last result of patching. This means that root can be anything (not even a DOM node, think of vDOM). As only the caller knows what `patchFn` does, the caller is the one to decide on how to cleanup. Trying a brute-force approach might have unexpected effects and can be confusing for users.